lib/utils/data_service.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:convert';
+import 'package:http/http.dart';
+
+class DataService {
+  static const String apiKey = '4bff35345ac2e30031ea2f22b88d7ff6';
+
+  Future<Map> fetchCurrentWeatherData(String city) async {
```

Comment:
It's fine for this little toy example, but in a real app you'd probably want to have this return a specific object (something like `WeatherData`) instead of a generic Map. The `fetchCurrentWeatherData` method would then parse the relevant data out of that map internally and instantiate the WeatherData object with it. That gives you a little more type safety because the WeatherData will tell everybody exactly what data it contains. With a map, you kind of have to know the keys to get the data out of it.